#!/bin/sh /usr/share/dpatch/dpatch-run

## 81_Make_config.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add Make.config to compile debug versions.

@DPATCH@
diff -urNad vdr-1.6.0~/Make.config vdr-1.6.0/Make.config
--- vdr-1.6.0~/Make.config	1970-01-01 01:00:00.000000000 +0100
+++ vdr-1.6.0/Make.config	2008-04-13 12:11:55.000000000 +0200
@@ -0,0 +1,28 @@
+#
+# The following compiler settings are required for Debian builds and will be
+# included by the Makefile of VDR and most plugins.
+#
+
+CFLAGS   = -g -Wall
+CXXFLAGS = -g -Wall -Woverloaded-virtual -Wno-parentheses
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+    CXXFLAGS += -O0
+    CFLAGS += -O0
+else
+    CXXFLAGS += -O2
+    CFLAGS += -O2
+endif
+
+ifndef BUILDVDR
+    # Required for plugins only, not for VDR itself
+    CXXFLAGS += -fPIC
+    CFLAGS += -fPIC
+endif
+
+ifdef VDRDEBUG
+    # Required for special debug build of VDR, where plugins are 
+    # not unloaded at runtime to give valgrind a chance to detect
+    # memory leaks in the plugin libs
+    DEFINES += -DVDRDEBUG
+endif
